src/Symfony/Component/HttpFoundation/Session/Storage/Handler/AbstractSessionHandler.php

```diff
@@ -121,30 +123,8 @@ public function destroy($sessionId)
             if (!$this->sessionName) {
                 throw new \LogicException(sprintf('Session name cannot be empty, did you forget to call "parent::open()" in "%s"?.', \get_class($this)));
             }
-            $sessionCookie = sprintf(' %s=', urlencode($this->sessionName));
-            $sessionCookieWithId = sprintf('%s%s;', $sessionCookie, urlencode($sessionId));
-            $sessionCookieFound = false;
-            $otherCookies = array();
-            foreach (headers_list() as $h) {
-                if (0 !== stripos($h, 'Set-Cookie:')) {
-                    continue;
-                }
-                if (11 === strpos($h, $sessionCookie, 11)) {
-                    $sessionCookieFound = true;
-
-                    if (11 !== strpos($h, $sessionCookieWithId, 11)) {
-                        $otherCookies[] = $h;
-                    }
-                } else {
-                    $otherCookies[] = $h;
-                }
-            }
-            if ($sessionCookieFound) {
-                header_remove('Set-Cookie');
-                foreach ($otherCookies as $h) {
-                    header($h, false);
-                }
-            } else {
+            $cookie = HeaderUtils::popSessionCookie($this->sessionName, $sessionId);
+            if (null === $cookie) {
                 setcookie($this->sessionName, '', 0, ini_get('session.cookie_path'), ini_get('session.cookie_domain'), ini_get('session.cookie_secure'), ini_get('session.cookie_httponly'));
```

Comment:
no need for samesite here also?

Reply:
This sets the session id to an empty value in the cookie, meaning the session won't work. I don't think adding SameSite here would add anything.

Looking at this some more, isn't this code supposed to remove the cookie entirely? In that case the value `0` is incorrect, that should be e.g. `time() - 3600`.

Reply:
> isn't this code supposed to remove the cookie entirely

indeed, my bad

> the value 0 is incorrect

I don't understand why O is invalid here. It's a timestamp in the past that's fine for removing a cookie AFAIK

Reply:
From [the manual on setcookie](https://secure.php.net/manual/en/function.setcookie.php):

> If set to 0, or omitted, the cookie will expire at the end of the session (when the browser closes).

Reply:
oups, would you mind opening a PR on 3.4 where this has been introduced?

Reply:
No problem, I'll do that later.

Reply:
Do we actually need a calculated value here, or wouldn't `1` be sufficient?

Reply:
I'll test it locally to be sure, but it seems to me that `1` would work :+1:

Reply:
So I tested this locally and the results are not as expected. When I send a cookie with an empty value and an `expires` of 0 it actually _does_ delete the cookie. Turns out further down the manual page it says:

> If the value argument is an empty string, or FALSE, and all other arguments match a previous call to setcookie, then the cookie with the specified name will be deleted from the remote client. This is internally achieved by setting value to 'deleted' and expiration time to one year in past.

Which is actually only partly true from what I've seen as:

- It doesn't set the expiration time to one year in the past, it sets it to 1970-01-01 00:00:01
- The value of `$httponly` doesn't seem to matter; I've a cookie was set with http `false` and you delete it with `http` true it gets deleted all the same, even though the documentation suggests it would not
- If I create a cookie with SameSite=lax and remove it without any samesite option, it still deletes the cookie

It does however always send a `Set-Cookie` header, it's just that it doesn't get picked up by the browser as a command to delete the previous cookie.

I've only tested in chrome, so the actual behaviour of when the header is honoured and when not might vary between browsers as well.

Maybe it would be best to add `ini_get('session.cookie_samesite')` to the call if it's PHP >= 7.3 just to be sure?

Reply:
I've added the `samesite` parameter for PHP >= 7.3 just to be sure.